python/src/aac/context/language_error.py

```diff
@@ -8,3 +8,23 @@ class LanguageError(Exception):
     """A base class representing a language error condition."""
     message: str = attrib(validator=validators.instance_of(str))
     location: Optional[str] = attrib(validator=validators.optional(validators.instance_of(str)))
+
+    def __init__(self, message: str, location: str):
+        """
+        An init constructor for the Language Error Class.
+
+        Args:
+            message(str): The error message.
+            location(str): The source location of the error.
+        """
+        self.message = message
+        self.location = location
```

Comment:
Should still provide the type hints here. 
```suggestion
        self.message: str = message
        self.location: Optional[str] = location
```

Reply:
Though, we need to discuss if we really want the location portion to be an optional attribute for the error. That is what allows for us to pass `None` in the `location` entries of a `LanguageError`. Do we really want a `None` or do we want a paired down `location` of at minimum returning the file that triggered the error?

Reply:
If we have None as location, we could instead give it a string like "No Location" or something, and remove the optional from the type.

Reply:
Also, I dont think it needs the type hints because its not creating the class variable, its just setting them.  It has type hints above in the class body.

Reply:
> If we have None as location, we could instead give it a string like "No Location" or something, and remove the optional from the type.

So, meaning where we are actually raising the errors if no applicable location provide the "No Location" notice instead?